enhancements/machine-api/azure-ultra-disks.md

```diff
@@ -0,0 +1,261 @@
+- if `UltraSSDCapability` is `Enabled` then the `UltraSSDEnabled` Additional capability will be set to `true` for the Machine, which will allow Ultra Disks attachments in both ways (via Data Disks and via PVCs)
+- if `UltraSSDCapability` is `Disabled` then the `UltraSSDEnabled` Additional capability will be set to `false` for the Machine, which will disallow Ultra Disks attachements in both ways  (via Data Disks and via PVCs)
+- if `UltraSSDCapability` is omitted, the provider may enable the `UltraSSDEnabled` capability depending on whether any Ultra Disks are specified as Data Disks
+
+```go
+type AzureMachineProviderSpec struct {
+  // Existing fields will not be modified
+  ...
+
+  // UltraSSDCapability enables or disables Azure UltraSSD capability for a virtual machine.
+  // When omitted, the platform may enable the capability based on the configuration of data disks.
+  // +kubebuilder:validation:Enum:="Enabled";"Disabled"
+  // +optional
+  UltraSSDCapability AzureUltraSSDCapabilityState `json:"ultraSSDCapability,omitempty"`
+}
+
+// AzureUltraSSDCapabilityState defines the different states of an UltraSSDCapability
+type AzureUltraSSDCapabilityState string
+
+// These are the valid AzureUltraSSDCapabilityState states.
+const (
+  // "AzureUltraSSDCapabilityTrue" means the Azure UltraSSDCapability is Enabled
+  AzureUltraSSDCapabilityTrue AzureUltraSSDCapabilityState = "Enabled"
+  // "AzureUltraSSDCapabilityFalse" means the Azure UltraSSDCapability is Disabled
+  AzureUltraSSDCapabilityFalse AzureUltraSSDCapabilityState = "Disabled"
+)
+```
+
+### Implementation Details/Notes/Constraints
+
+N/A
+
+### Risks and Mitigations
+
+N/A
+
+## Design Details
+
+### Open Questions
+
+The ability to attach Ultra Disks to Pods (and Machines) via Persistent Volumes Claims [is not yet supported](https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/1852) in the upstream Cluster API Provider for Azure (CAPZ).
+
+I ([@damdo](https://github.com/damdo)) volunteered to propose an implementation of this feature in the upstream Cluster API Provider for Azure (CAPZ).
+But there is a question on whether the to-be-proposed upstream API Extension for Ultra Disks as Persistent Volumes (which will likely mirror what it's being proposed here) ends up differing in the final accepted design and implementation from the one proposed for this downstream forum (OCP).
+
+### Test Plan
+TBD 
+
+### Graduation Criteria
+The addition of API fields to Machine API implies that the feature is GA from the beginning, no graduation criteria are required.
+
+#### Dev Preview -> Tech Preview
+
+N/A
+
+#### Tech Preview -> GA
+
+N/A
+
+#### Removing a deprecated feature
+
+No features will be removed as a part of this proposal.
+
+### Upgrade / Downgrade Strategy
+
+Existing clusters being upgraded will not be configured for the new feature and will continue to use the existing standard interface type.
```

Comment:
Standard interface type? This section might want a reword as I'm not sure interface is the right wording here

Reply:
Ahh good catch, this was a copy/pasta, the text was not supposed to be that one.
Paragraph updated.